68 refactor pile properties in line with pilecore api v3 definition

tlukkezen opened the pull request:
closes #68

RDWimmers commented on src/pypilecore/common/piles/geometry/components/round.py:
```diff
@@ -0,0 +1,435 @@
+                patches.Rectangle(
+                    (self.cross_section_bounds[0], z_offset),
+                    self.diameter,
+                    height,
```
if a `'primary_dimension': {'length': float}` is given can we add that to the plot as well.

For example when `height_base=1`. 
![image](https://github.com/user-attachments/assets/22e75579-f52e-4d13-b877-b11c9b546820)

tlukkezen replied:
fixed

RDWimmers reviewed (approved):
Maybe we can give an overview of what the different dimensions mean. We created a already something like that, right?


Also need to update the docs -> https://github.com/cemsbv/py-pilecore/blob/b4e5563d4b7fbe915ccf25acbc799547f019410a/docs/tree/reference.rst?plain=1#L57

tlukkezen commented on pyproject.toml:
```diff
@@ -81,7 +81,7 @@ disallow_untyped_calls = true
 no_implicit_reexport = true
 strict_equality = true
-install_types = true
+ignore_missing_imports = true
```
docs are updated, mypy settings fixed, image can be fixed in another PR

RDWimmers commented:
![image](https://github.com/user-attachments/assets/d71bcd80-5bab-4878-981f-7e9f4897ffb3)

tlukkezen commented:
# Draft Release notes:


This release connects py-pilecore to the freshly release PileCore-API v3.0.
### A new PileProperties object
One of the main breaking changes in the API is the definition of a foundation pile object. This is also reflected in py-pilecore by adding a brand new `PileProperties` object, which holds the `geometry` and `pile_type` information. This `PileProperties` object can be created with the function `create_basic_pile()`:
```diff
    create_basic_pile(
        pile_name=pile_name,
+        main_type=main_type,
        specification=specification,
        installation=installation,
        pile_shape=pile_shape,
        height_base=height_base,
+        core_secondary_dimension=core_secondary_dimension,
+        core_tertiary_dimension=core_tertiary_dimension,
+        base_secondary_dimension=base_secondary_dimension,
+        base_tertiary_dimension=base_tertiary_dimension,
+        core_diameter=core_diameter,
+        base_diameter=base_diameter,
+        pile_material=pile_material,
+        custom_material=custom_material,
        settlement_curve=settlement_curve,
        adhesion=adhesion,
        alpha_p=alpha_p,
        alpha_s_clay=alpha_s_clay,
        alpha_s_sand=alpha_s_sand,
        beta_p=beta_p,
        pile_tip_factor_s=pile_tip_factor_s,
        is_auger=is_auger,
        is_low_vibrating=is_low_vibrating,
        negative_fr_delta_factor=negative_fr_delta_factor,
    )
```
Most of the input arguments were already known as arguments for creating the API payload, but there are also some new arguments, as marked above.
The changes in the pile dimensions are probably standing out the most. They reflect the changes in the pile geometry model in the API, which is made to be more modular. In the new model, we always require a *core* component of the pile, and allow extra components to be formed around it (like a *Russian doll* model). Therefore we now require the dimensions of the *core*, while the *base* dimensions are optional.
It can also be seen that the pile can also be assigned a *material*, which sets properties like the *E-modulus* and the *color* for plotting purposes.
The pile and its separete components can be visualized with plotting methods, so that you can inspect the pile shape in `py-pilecore` before sending it to the API.
Furthermore it is important to notice that we also renamed and simplified the definition of the `default_pile` (previously `pile_type_specification`). Not just the name is shorter: We also changed the required input for creating the pile-type, so be sure to check out the new pile_type tables!
